integration-tests/capacity/replenishment.test.ts

```diff
@@ -0,0 +1,174 @@
+      const payload = JSON.stringify({ changeType: 1,  fromId: 1, objectId: 2 })
+      // const call = ExtrinsicHelper.api.tx.messages.addOnchainMessage(null, schemaId, payload);
+      const call = ExtrinsicHelper.addOnChainMessage(stakeKeys, schemaId, payload);
+
+      // confirm that we start with a full tank
+      let remainingCapacity = (await getRemainingCapacity(stakeProviderId)).toBigInt();
+      assert.equal(totalCapacity, remainingCapacity);
+
+      await call.payWithCapacity(-1);
+      remainingCapacity = (await getRemainingCapacity(stakeProviderId)).toBigInt();
+      assert(totalCapacity > remainingCapacity);
+
+      // one more txn to deplete capacity more so this current remaining is different from when
+      // we submitted the first message.
+      await call.payWithCapacity(-1);
+      const newEpochBlock = await getNextEpochBlock();
+      await ExtrinsicHelper.run_to_block(newEpochBlock);
+
+      // this should cause capacity to be refilled and then deducted by the cost of one message.
+      await call.payWithCapacity(-1);
+      let newRemainingCapacity = (await getRemainingCapacity(stakeProviderId)).toBigInt();
+
+      // this should be the same as after sending the first message, since it is the first message after
+      // the epoch.
+      assert.equal(remainingCapacity, newRemainingCapacity);
+    });
+  });
+
+  describe("Capacity is not replenished", function() {
+    it("if out of capacity and last_replenished_at is <= current epoch", async function() {
+      let [stakeKeys, stakeProviderId] = await createAndStakeProvider("NoSend", 3n*1000n*1000n);
+      let payload = JSON.stringify({ changeType: 1,  fromId: 1, objectId: 2 })
+      let call = ExtrinsicHelper.addOnChainMessage(stakeKeys, schemaId, payload);
+      const expectedApproxCost = 1891000n;
+
+      // run until we can't afford to send another message.
+      await call.payWithCapacity(-1);
+      let remainingCapacity = (await getRemainingCapacity(stakeProviderId)).toBigInt();
+      assert(remainingCapacity < expectedApproxCost);
+
+      let nextEpochBlock = await getNextEpochBlock();
+
+      // mine until just before epoch rolls over
+      await ExtrinsicHelper.run_to_block(nextEpochBlock - 1);
+
+      // TODO: it's weird that we can run one more capacity txn before running out.
+      await call.payWithCapacity(-1)
+
+      await assert.rejects(
+        call.payWithCapacity(-1), {name: "RpcError", message: "1010: Invalid Transaction: Inability to pay some fees , e.g. account balance too low"});
+    });
+  });
+
+  describe("Regression test: when user attempts to stake tiny amounts before provider's first message of an epoch,", function() {
+    it("provider is still replenished and can send a message", async function(){
+      const providerStakeAmt = 2n*1000n*1000n;
+      const userStakeAmt = 1n*1000n*1000n;
+      const userIncrementAmt = 1000n;
+      const expectedApproxCost = 1891000n;
```

Comment:
We cannot now for capacity transactions, but we will probably add an API.

Reply:
Capacity fee will be based on what's in GetStableWeights which won't change as much over time, however, I guess since I'm doing the same call repeatedly I could just get it by subtracting remaining capacity from before the first call. I'll do that instead.